includes/quiz-submission/submission/repositories/class-submission-comments-based-repository.php

```diff
@@ -0,0 +1,187 @@
+<?php
+/**
+ * File containing the Submission_Comments_Based_Repository class.
+ *
+ * @package sensei
+ */
+
+namespace Sensei\Quiz_Submission\Submission\Repositories;
+
+use DateTime;
+use DateTimeInterface;
+use RuntimeException;
+use Sensei\Quiz_Submission\Submission\Models\Submission;
+use Sensei_Utils;
+use WP_Comment;
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+/**
+ * Class Submission_Comments_Based_Repository.
+ *
+ * @since $$next-version$$
+ */
+class Submission_Comments_Based_Repository implements Submission_Repository_Interface {
```

Comment:
I noticed you put the "specification" in the middle of the name.
I think it would be better to put it as a prefix: it will help us to find a difference quicker. 
When we use the name of the class directly, we work in the submission's context: the file lies in the submission subdirectory (and have the Submission namespace) and all the repositories start with the same word, which means we need to read further :)

Reply:
I'm ok with both solutions, so no problem. 🙂 The classes have been renamed in bdb61b61271082fb83c8db2d6d1feaf4c7660d30.